### PR TITLE
Changes made and updated the HTTP response as per the TEST cases results

### PR DESCRIPTION
**Pull request by Azher2Ali**
Changes made and updated the HTTP response as per the TEST cases results

### CODE REVIEW

**Review by joneubank — CHANGES_REQUESTED** [review]
Thanks for updating the unit tests as well...

For the test cases where the API Key is rejected, its good that we are responding with `OK` but we should also be checking that the response includes `valid: false`

File: src/test/java/bio/overture/ego/controller/ApiKeyControllerTest.java [hunk 1]
```diff
@@ -358,7 +358,10 @@ public void checkInvalidApiKey() {
     val response = initStringRequest().endpoint("o/check_api_key").body(params).post();
 
     val statusCode = response.getStatusCode();
-    assertEquals(HttpStatus.UNAUTHORIZED, statusCode);
+    /*assertEquals(HttpStatus.BAD_REQUEST, statusCode);
+    assertEquals("false", response.getStatusCode());*/
```
Comment (joneubank) [inline src/test/java/bio/overture/ego/controller/ApiKeyControllerTest.java:362]: Please remove unused comments.

File: src/test/java/bio/overture/ego/controller/ApiKeyControllerTest.java [hunk 2]
```diff
@@ -358,7 +358,10 @@ public void checkInvalidApiKey() {
...
+      
+      assertEquals(HttpStatus.BAD_REQUEST, "false");*/
```
Comment (joneubank) [inline src/test/java/bio/overture/ego/controller/ApiKeyControllerTest.java:364]: We should be doing tests on the structure we expect to see:

statusCode = OK
body.valid = false

File: src/test/java/bio/overture/ego/controller/TokensOnPermissionsChangeTest.java [hunk 3]
```diff
@@ -102,7 +102,7 @@ public void deletePermissionFromUser_ExistingToken_RevokeSuccess() {
         initStringRequest(tokenHeaders).endpoint("/o/check_api_key?apiKey=%s", apiKey).post();
 
     // Should be revoked
-    assertEquals(checkTokenAfterDeleteResponse.getStatusCode(), HttpStatus.UNAUTHORIZED);
+    assertEquals(checkTokenAfterDeleteResponse.getStatusCode(), HttpStatus.OK);
```
Comment (joneubank) [inline src/test/java/bio/overture/ego/controller/TokensOnPermissionsChangeTest.java:105]: all the checks for revoked token in this test file are incomplete. a response with status OK can be either valid or invalid. We also need to check if the response has the property `valid: false`

**Review by joneubank — CHANGES_REQUESTED** [review]
This needs attention. The code does not compile as written and does not test for the case we expect. I left comments with additional details.

**Comment by joneubank** [comment]
I've been doing some thinking about these changes... I think we should remove the changes from the other PR instead of pushing through with test case updates. The changes this causes will affect consuming applications.. this is a breaking change that I didn't prepare for thoroughly enough.

If we are going to make these changes, we need to provide a new endpoint (perhaps versioned as v2) and have clearer documentation about the response type. We should also include the same "valid" property with `valid: true` on success.  All in all, this was poorly conceived and prepared for by myself and for now I think we should revert... I'll help prep a PR to pull these changes out.

**Comment by justincorrigible** [comment]
Closing as result of https://github.com/overture-stack/ego/pull/684

**Pull request closed without merging**